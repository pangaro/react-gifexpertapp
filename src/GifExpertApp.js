
import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

//import PropTypes from 'prop-types'

//const GifExpertApp = props => {
export const GifExpertApp = ({ defaultCategories = [] }) => {
    //const categories = [ 'pepe', 'pepa', 'pipa' ];

    //const [categories, setCategories] = useState([]);
    const [categories, setCategories] = useState( defaultCategories );
    /*const handleAdd = (e) => {
        //setCategories( [...categories, 'uno mas'] );
        setCategories( cats => [...categories, 'uno mas'] );
    }*/

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid
                            key={ category }
                            category={ category } 
                        />
                        //<li key={ category }> { category } </li>
                    ))
                }
            </ol>
        </>
    )
}

/*GifExpertApp.propTypes = {

}*/

export default GifExpertApp;
