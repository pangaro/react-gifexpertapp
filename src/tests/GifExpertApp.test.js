import React from 'react';
import GifExpertApp from "../GifExpertApp";
import { shallow } from 'enzyme';


describe('pruebas en <GifEspertApp />', () => {
    

    test('debe mostrarse correctamente', () => {

        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => {
       
        const categories = ['goku', 'vegeta'];

        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
    
})
