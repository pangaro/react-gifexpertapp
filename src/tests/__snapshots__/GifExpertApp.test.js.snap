// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifEspertApp /> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="goku"
      key="goku"
    />
    <GifGrid
      category="vegeta"
      key="vegeta"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en <GifEspertApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
